shell/platform/darwin/common/framework/Source/FlutterCodecs.mm

```diff
@@ -105,9 +106,9 @@ - (id)decode:(NSData*)message {
       [expandedMessage replaceBytesInRange:NSMakeRange(message.length + 1, 1) withBytes:&end];
       message = expandedMessage;
     }
-    decoded = [NSJSONSerialization JSONObjectWithData:message options:0 error:nil];
+    decoded = [NSJSONSerialization JSONObjectWithData:message options:0 error:&error];
   }
-  NSAssert(decoded, @"Invalid JSON message, decoding failed");
+  NSAssert(decoded, @"Invalid JSON message, decoding failed: %@", error);
```

Comment:
Similar to what's already done in encode https://github.com/flutter/engine/blob/ed73d40a8c9340dc0d9c2e2a859d02b876dc9858/shell/platform/darwin/common/framework/Source/FlutterCodecs.mm#L82